vistautils/key_value.py

```diff
@@ -733,16 +807,28 @@ def __exit__(self, exc_type, exc_value, traceback):
 
 _CHAR_KEY_VALUE_SOURCE_SPECIAL_VALUES = {
     "zip": "_ZipCharFileKeyValuesSource",
-    "file-map": "DocIdToFileMapCharKeyValueSource",
+    "file-map": "_PathMappingCharKeyValueSource",
+}
+
+_BYTE_KEY_VALUE_SOURCE_SPECIAL_VALUES = {
+    "zip": "_ZipBytesFileKeyValuesSource",
+    "file-map": "_PathMappingBytesKeyValueSource",
 }
 
 
 def _doc_id_source_from_params(params: Parameters) -> KeyValueSource[str, str]:
     return KeyValueSource.from_doc_id_to_file_map(params.existing_file("path"))
 
 
+def _doc_id_binary_source_from_params(params: Parameters) -> KeyValueSource[str, bytes]:
+    return KeyValueSource.binary_from_doc_id_to_file_map(params.existing_file("path"))
+
+
 def char_key_value_linear_source_from_params(
-    param_name: str, params: Parameters, *, eval_context: Optional[Dict] = None
+    params: Parameters,
+    *,
+    input_namespace: str = "input",
+    eval_context: Optional[Dict] = None,
```

Comment:
```suggestion
    eval_context: Optional[Mapping[Any, Any]] = None,
```
for all the functions that take an `eval_context`.

Reply:
I also think that as a rule, we should disallow the implicit `Any` in annotations.

Reply:
Added https://github.com/isi-vista/vistautils/issues/86